packages/react/src/components/button.tsx

```diff
@@ -218,23 +222,27 @@ export function UploadButton<
     );
     if (customContent) return customContent;
 
-    if (state === "readying") return "Loading...";
-
-    if (state !== "uploading") {
+    if (state === "readying") {
+      return "Loading...";
+    } else if (state === "uploading") {
+      if (uploadProgress === 100) {
+        return <Spinner />;
+      } else {
+        return (
+          <span className="z-50">
+            <span className="block group-hover:hidden">{uploadProgress}%</span>
+            <Cancel className="hidden size-4 group-hover:block" />
+          </span>
+        );
+      }
+    } else {
+      // Default case: "ready" or "disabled" state
       if (mode === "manual" && files.length > 0) {
         return `Upload ${files.length} file${files.length === 1 ? "" : "s"}`;
+      } else {
+        return `Choose File${inputProps.multiple ? `(s)` : ``}`;
       }
-      return `Choose File${inputProps.multiple ? `(s)` : ``}`;
     }
```

Comment:
quite a bit of nestled code here after return statements. do you prefer this style?

Reply:
No specific preference, would a switch be better?

```tsx
const renderButton = () => {
    const customContent = contentFieldToContent(
      $props.content?.button,
      styleFieldArg,
    );
    if (customContent) return customContent;

    switch (state) {
      case "readying": {
        return "Loading...";
      }
      case "uploading": {
        if (uploadProgress === 100) return <Spinner />;
        return (
          <span className="z-50">
            <span className="block group-hover:hidden">{uploadProgress}%</span>
            <Cancel className="hidden size-4 group-hover:block" />
          </span>
        );
      }
      case "disabled":
      case "ready":
      default: {
        if (mode === "manual" && files.length > 0) {
          return `Upload ${files.length} file${files.length === 1 ? "" : "s"}`;
        }
        return `Choose File${inputProps.multiple ? `(s)` : ``}`;
      }
    }
  };
  ```

Reply:
yea either that or just remove the `else` / `else if` cause the branch before returns anyways.